doc/read-the-docs-site/architecture/key-concepts.rst

```diff
@@ -1,6 +1,242 @@
-Key concepts
-============
+Marconi Key concepts
+====================
 
-We'll explain concepts such as ``Indexer``, ``Coordinator``, ``Worker``, ``Transformer``, ``Preprocessor``, ``Queryable``, ``IsSync``, ``IsIndex``, ``chain-sync protocol``, etc.
+We introduce here the key concepts of Marconi, how they are related and how they
+map to the architecture
 
-Coming eventually!
+Indexer
+-------
+
+The key component of Marconi is an ``indexer``.
+An indexer can be any type that implement the ``IsIndex`` typeclass. The
+definition of ``IsIndex`` is (very close) to this implementation:
+
+.. code-block:: haskell
+   class (Monad m) => IsIndex m event indexer where
+
+     index
+       :: (Eq (Point event))
+       => Timed (Point event) (Maybe event)
+       -> indexer event
+       -> m (indexer event)
+
+    // …
+
+    rollback :: (Ord (Point event)) => Point event -> indexer event -> m (indexer event)
+
+Let's discuss the type variables first:
+
+- ``m`` is the monadic context of the indexer.
+- ``event`` defines the input type of the indexer, we usually stores these
+  events, but an indexer can also decide to preprocess them and store a
+  different type.
```

Comment:
It can be done without preprocessing, by putting logic into the `index` function. Alhough, I understand that using _preprocess_ is confusing. Changing it to process.